client/packages/programs/src/JsonForms/components/Search/SearchWithDocumentSource.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from 'react';
+import { ControlProps } from '@jsonforms/core';
+import { get as extractProperty } from 'lodash';
+import {
+  useTranslation,
+  Box,
+  FormLabel,
+  Typography,
+} from '@openmsupply-client/common';
+import {
+  FORM_LABEL_COLUMN_WIDTH,
+  FORM_INPUT_COLUMN_WIDTH,
+} from '../../common/styleConstants';
+import { z } from 'zod';
+import { useZodOptionsValidation } from '../../common/hooks/useZodOptionsValidation';
+import { usePatientStore } from 'packages/programs/src/hooks';
+import { useEncounter } from '../../../api';
+import { RegexUtils, JSXFormatters } from '@openmsupply-client/common';
+
+const Options = z
+  .object({
+    source: z.enum(['document']),
+    /**
+     * Which pre-defined query to use (in useSearchQueries)
+     */
+    document: z.enum(['patient', 'encounter']),
+    /**
+     * Path in the specified document to extract
+     */
+    path: z.string(),
+    /**
+     * Pattern for formatting selected result (as above)
+     */
+    displayString: z.string().optional(),
+    /**
+     * List of fields to save in document data (from selected item object)
+     */
+    saveFields: z.array(z.string()).optional(),
+  })
+  .strict();
+
+type Options = z.infer<typeof Options>;
+
+const { formatTemplateString } = RegexUtils;
+const { replaceHTMLlineBreaks } = JSXFormatters;
+
+export const SearchWithDocumentSource = (props: ControlProps) => {
+  const {
+    errors: formErrors,
+    path,
+    handleChange,
+    label,
+    uischema,
+    visible,
+  } = props;
+  const { errors: zErrors, options } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+  const { currentPatient = {} } = usePatientStore();
+
+  const {
+    data: currentEncounter,
+    isLoading,
+    isError,
+  } = useEncounter.document.byId(useEncounter.utils.idFromUrl());
```

Comment:
Current encounter only available when we're in an Encounter form, so that the encounter ID can be fetched from URL. Will report an error (see below) if the specified document (e.g. "encounter") is unavailable in the current context.

Reply:
as in comment above you could just use the "current" document

Reply:
Yeah, that would be good, but wasn't sure how to access the generic "current" document. Any suggestions?